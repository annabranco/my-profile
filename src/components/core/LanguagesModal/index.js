import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { bool, func } from 'prop-types';
import AppButton from '../../elements/AppButton';
import {
  allLanguagesSelector,
  currentLanguageSelector,
  languagesModelTextsSelector
} from '../../../redux/selectors';
import { changeLanguage } from '../../../redux/actions/languages';
import {
  BackgroundOverlay,
  CheckBox,
  CheckBoxArea,
  Flag,
  FlagsWrapper,
  Image,
  Label,
  ModalWrapper,
  Text
} from './styles';
import { getFlagURL } from '../../../utils/icons';

const updateLanguageSettings = (language, hideForever) =>
  localStorage.setItem(
    "Anna Branco's professional profile",
    JSON.stringify({
      language,
      hideLanguagesModalForever: hideForever
    })
  );

export const onChangeLanguage = (
  currentLang,
  newLang,
  hideForever,
  dispatch
) => {
  if (newLang !== currentLang) {
    dispatch(changeLanguage(newLang));
    updateLanguageSettings(newLang, hideForever);
  }
};

const LanguagesModal = ({
  hideForever,
  onCloseLanguageModal,
  toggleBlockLangModal
}) => {
  const texts = useSelector(languagesModelTextsSelector);
  const languages = useSelector(allLanguagesSelector);
  const languageSelected = useSelector(currentLanguageSelector);

  const dispatch = useDispatch();

  const onClickClose = () => {
    onCloseLanguageModal(hideForever);
    updateLanguageSettings(languageSelected, hideForever);
  };

  const onCLickFlag = event =>
    onChangeLanguage(
      languageSelected,
      event.currentTarget.lang,
      hideForever,
      dispatch
    );

  return (
    <BackgroundOverlay>
      <ModalWrapper>
        <FlagsWrapper>
          {languages.map(item => (
            <Flag
              active={item.active}
              aria-label={item.language}
              key={item.languageCode}
              lang={item.languageCode}
              languageSelected={languageSelected}
              onClick={onCLickFlag}
              role="button"
              tabIndex={0}
            >
              <Label>{item.language}</Label>
              <Image
                alt=""
                lang={item.languageCode}
                languageSelected={languageSelected}
                src={getFlagURL({
                  country: item.flagCode,
                  style: 'flat3d',
                  size: 'small'
                })}
              />
            </Flag>
          ))}
        </FlagsWrapper>

        <Text>{texts.changeUpperBar}</Text>
        <CheckBoxArea role="checkbox" onClick={toggleBlockLangModal}>
          <CheckBox
            className="languages__checkbox"
            checked={hideForever}
            id="checkLang"
            name="hideForever"
            onClick={toggleBlockLangModal}
            type="checkbox"
            readOnly
          />
          <label htmlFor="blockLang">{texts.chekboxText}</label>
        </CheckBoxArea>
        <AppButton buttonStyle="confirm" onClick={onClickClose} text="ok" />
      </ModalWrapper>
    </BackgroundOverlay>
  );
};

LanguagesModal.propTypes = {
  hideForever: bool.isRequired,
  onCloseLanguageModal: func.isRequired,
  toggleBlockLangModal: func.isRequired
};

export default LanguagesModal;
