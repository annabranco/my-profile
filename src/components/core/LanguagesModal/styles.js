import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlueDark,
  colorBlack,
  colorGrayNormal,
  fontSubtitle,
  fontTitleAlt
} from '../../../styles/theme';
import { NotDisplayed } from '../../../styles/global';

const activeFlag = css`
  opacity: 1;
  cursor: default;
`;

export const Text = styled.p`
  margin: 5px 0;
  font-family: ${fontSubtitle};
  font-size: 0.8rem;
  text-align: center;

  @media all and (min-width: 560px) {
    font-size: 1rem;
  }
`;
Text.displayName = 'Text';

export const BackgroundOverlay = styled.div`
  z-index: 3;
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(
    to bottom,
    ${rgba(colorWhite, 0.6)},
    ${rgba(colorBlueDark, 0.4)},
    ${rgba(colorBlueDark, 0.6)},
    ${rgba(colorBlack, 0.9)}
  );
  height: 100vh;
  width: 100%;
`;
BackgroundOverlay.displayName = 'Background Overlay';

export const CheckBox = styled.input`
  margin-right: 20px;
`;
CheckBox.displayName = 'CheckBox';

export const CheckBoxArea = styled(Text)`
  margin: 30px 0 10px;
`;
CheckBoxArea.displayName = 'CheckBox Area';

export const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border: 1px solid;
  border-radius: 40px 0;
  background-image: linear-gradient(
    to bottom left,
    ${colorWhite},
    #e6f4f4,
    ${colorBlueLight}
  );
  height: 280px;
  width: 340px;
  padding: 20px;
  box-shadow: 2px 2px 10px 2px ${rgba(colorGrayNormal, 0.7)};

  @media all and (min-width: 560px) {
    border-radius: 50px 0;
    height: 260px;
    width: 480px;
  }
`;
ModalWrapper.displayName = 'Modal Wrapper';

export const Flag = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  outline: none;
  margin: 10px;
  cursor: pointer;

  ${props =>
    !props.active &&
    css`
      ${NotDisplayed};
    `}
`;
Flag.displayName = 'Flag';

export const FlagsWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 70px;
  margin: 10px auto 0;

  @media all and (min-width: 560px) {
    margin: 20px auto 0;
  }
`;
FlagsWrapper.displayName = 'Flags Wrapper';

export const Image = styled.img`
  opacity: 0.5;
  margin-top: -10px;
  width: 100%;

  ${props =>
    props.lang === props.languageSelected &&
    css`
      ${activeFlag}
    `}
`;
Image.displayName = 'Image';

export const Label = styled.div`
  margin-bottom: 3px;
  font-family: ${fontTitleAlt};
  font-size: 0.8rem;
  text-align: center;
`;
Label.displayName = 'Label';
