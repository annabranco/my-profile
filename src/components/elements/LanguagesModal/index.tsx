import { Dispatch, ReactElement } from 'react';
import * as React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { bool, func } from 'prop-types';
import { ILanguageCode, ILanguage } from '../../../types/interfaces';
import {
  allLanguagesSelector,
  currentLanguageSelector,
  languagesModelTextsSelector
} from '../../../redux/selectors';
import { changeLanguage } from '../../../redux/actions/languages';
import { getFlagURL } from '../../../utils/icons';
import AppButton from '../AppButton';
import {
  BackgroundOverlay,
  CheckBox,
  CheckBoxArea,
  Flag,
  FlagsWrapper,
  Image,
  Label,
  ModalWrapper,
  Text
} from './styles';

const updateLanguageSettings = (language: string, hideForever: boolean): void =>
  localStorage.setItem(
    "Anna Branco's professional profile",
    JSON.stringify({
      language,
      hideLanguagesModalForever: hideForever
    })
  );

interface DispatchAction {
  payload: string;
  type: string;
}

export const onChangeLanguage = (
  currentLang: string,
  newLang: string,
  hideForever: boolean,
  dispatch: Dispatch<DispatchAction>
): void => {
  if (newLang !== currentLang) {
    dispatch(changeLanguage(newLang));
    updateLanguageSettings(newLang, hideForever);
  }
};

interface Props {
  hideForever: boolean;
  onCloseLanguageModal: () => void;
  toggleBlockLangModal: () => void;
}

const LanguagesModal = ({
  hideForever,
  onCloseLanguageModal,
  toggleBlockLangModal
}: Props): ReactElement => {
  const texts = useSelector(languagesModelTextsSelector);
  const languages: ILanguage[] = useSelector(allLanguagesSelector);
  const languageSelected: ILanguageCode = useSelector(currentLanguageSelector);

  const dispatch = useDispatch();

  const onClickClose = (event: React.MouseEvent): void => {
    event.stopPropagation();
    onCloseLanguageModal();
    updateLanguageSettings(languageSelected, hideForever);
  };

  const onCLickFlag = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ): void => {
    event.stopPropagation();
    onChangeLanguage(
      languageSelected,
      event.currentTarget.lang,
      hideForever,
      dispatch
    );
  };

  return (
    <BackgroundOverlay>
      <ModalWrapper data-e2e-id="languagesModal">
        <FlagsWrapper>
          {languages.map(item => (
            <Flag
              active={item.active}
              aria-label={item.language}
              data-e2e-id={`langModal-flag_${item.flagCode}`}
              key={item.languageCode}
              lang={item.languageCode}
              languageSelected={languageSelected}
              onClick={onCLickFlag}
              role="button"
              tabIndex={0}
            >
              <Label>{item.language}</Label>
              <Image
                alt=""
                lang={item.languageCode}
                languageSelected={languageSelected}
                src={getFlagURL({
                  country: item.flagCode,
                  style: 'flat3d',
                  size: 'small'
                })}
              />
            </Flag>
          ))}
        </FlagsWrapper>

        <Text data-e2e-id="langModal-instructionsText">
          {texts.changeUpperBar}
        </Text>
        <CheckBoxArea role="checkbox" onClick={toggleBlockLangModal}>
          <CheckBox
            checked={hideForever}
            className="languages__checkbox"
            data-e2e-id="langModal-checkbox"
            id="checkLang"
            name="hideForever"
            // onClick={toggleBlockLangModal}
            readOnly
            type="checkbox"
          />
          <label htmlFor="blockLang">{texts.chekboxText}</label>
        </CheckBoxArea>
        <AppButton
          buttonStyle="confirm"
          datae2eid="langModal-ok"
          onClick={onClickClose}
          text="ok"
        />
      </ModalWrapper>
    </BackgroundOverlay>
  );
};

LanguagesModal.propTypes = {
  hideForever: bool.isRequired,
  onCloseLanguageModal: func.isRequired,
  toggleBlockLangModal: func.isRequired
};

export default LanguagesModal;
