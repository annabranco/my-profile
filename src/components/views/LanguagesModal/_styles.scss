.background__overlay {
  z-index          : 3;
  position         : absolute;
  display          : flex;
  justify-content  : center;
  align-items      : center;
  background-image : linear-gradient(
    to bottom,
    rgba($white, 0.6),
    rgba($blue--dark, 0.4),
    rgba($blue--dark, 0.6),
    rgba($black, 0.9)
  );
  height : 100vh;
  width  : 100vw;
}

.languages__outer {
  display          : flex;
  flex-direction    : column;
  justify-content  : flex-start;
  border           : 1px solid;
  border-radius    : 40px 0;
  background-image : linear-gradient(
    to bottom left,
    $white,
    #e6f4f4,
    $blue--light
  );
  height     : auto;
  width      : 320px;
  padding    : 20px;
  box-shadow : 2px 2px 10px 2px rgba($gray-normal, 0.7);

  @media all and (min-width: 560px) {
    border-radius : 80px 0;
    width         : 560px;
  }
}

.languages__text {
  margin      : 5px 0;
  font-family : sans-serif;
  font-size   : 0.8rem;
  text-align  : center;

  @media all and (min-width: 560px) {
    font-size : 1rem;
  }
}

.languages__flags--outer {
  display               : grid;
  grid-template-columns : 1fr 1fr 1fr 1fr;
  grid-template-rows    : 70px;
  margin                : 10px auto 0;

  @media all and (min-width: 560px) {
    margin : 20px auto 0;
  }
}

.languages__flags--inner {
  display         : flex;
  flex-direction   : column;
  justify-content : center;
  align-items     : center;
  margin          : 10px;
  cursor          : pointer;
}

.languages__label {
  margin-bottom : 3px;
  font-size     : 0.8rem;
  text-align    : center;
}

.languages__flag {
  opacity    : 0.5;
  margin-top : -10px;
  width      : 100%;
}

.languages__flag--active {
  opacity : 1;
  cursor  : default;
}

.notYet {
  opacity : 0.2;
  filter  : grayscale(100%);
  cursor  : default;
}

.languages__checkbox {
  margin-right : 20px;
}

.languages__text--checkbox {
  margin : 30px 0 10px;
}

.languages__close {
  border           : 1px solid $black;
  border-radius    : 15px 0;
  margin           : 0 auto;
  background-image : linear-gradient(rgba($white, 0.2), rgba($blue--light, 0.6));
  width            : 100px;
  padding          : 5px 0;
  text-decoration  : none;
  font-family      : $font__title;
  font-size        : 1.2rem;
  cursor           : pointer;
  box-shadow       : 0 2px 4px 0 rgba($black, 0.4),
    inset 0 1px 5px 1px rgba($white, 0.7);

  &:hover {
    transform       : translate(0, 1px);
    text-decoration : none;
    box-shadow      : inset 0 2px 5px 1px rgba($black, 0.4);
  }
}
