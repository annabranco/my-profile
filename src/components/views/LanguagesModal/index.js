import React from 'react';
import { arrayOf, bool, func, string } from 'prop-types';
import AppButton from '../elements/AppButton';
import {
  BackgroundOverlay,
  CheckBox,
  CheckBoxArea,
  Flag,
  FlagsWrapper,
  Image,
  Label,
  ModalWrapper,
  Text
} from './styles';
import { languagesPropType, languagesTextPropType } from '../../../types/index';

const LanguagesModal = ({
  hideForever,
  languages,
  languageSelected,
  onChangeLanguage,
  onCloseLanguageModal,
  texts,
  toggleBlockLangModal
}) => (
  <BackgroundOverlay>
    <ModalWrapper>
      <FlagsWrapper>
        {languages.map(item => (
          <Flag
            active={item.active}
            aria-label={item.language}
            key={item.languageCode}
            lang={item.languageCode}
            languageSelected={languageSelected}
            onClick={onChangeLanguage}
            role="button"
            tabIndex={0}
          >
            <Label>{item.language}</Label>
            <Image
              alt=""
              lang={item.languageCode}
              languageSelected={languageSelected}
              src={`https://www.countryflags.io/${item.flagCode}/shiny/64.png`}
            />
          </Flag>
        ))}
      </FlagsWrapper>

      <Text>{texts.changeUpperBar}</Text>
      <CheckBoxArea>
        <CheckBox
          checked={hideForever}
          className="languages__checkbox"
          onClick={toggleBlockLangModal}
          type="checkbox"
        />
        {texts.chekboxText}
      </CheckBoxArea>
      <AppButton
        buttonStyle="confirm"
        onClick={onCloseLanguageModal}
        text="ok"
      />
    </ModalWrapper>
  </BackgroundOverlay>
);

LanguagesModal.propTypes = {
  hideForever: bool.isRequired,
  languages: arrayOf(languagesPropType).isRequired,
  languageSelected: string.isRequired,
  onChangeLanguage: func.isRequired,
  onCloseLanguageModal: func.isRequired,
  texts: languagesTextPropType.isRequired,
  toggleBlockLangModal: func.isRequired
};

export default LanguagesModal;
