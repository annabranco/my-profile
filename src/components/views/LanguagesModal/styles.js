import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlueDark,
  colorBlack,
  colorGrayNormal,
  fontSubtitle,
  fontTitleAlt
} from '../../../styles/theme';

export const BackgroundOverlay = styled.div`
  z-index: 3;
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(
    to bottom,
    ${rgba(colorWhite, 0.6)},
    ${rgba(colorBlueDark, 0.4)},
    ${rgba(colorBlueDark, 0.6)},
    ${rgba(colorBlack, 0.9)}
  );
  height: 100vh;
  width: 100vw;
`;

export const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border: 1px solid;
  border-radius: 40px 0;
  background-image: linear-gradient(
    to bottom left,
    ${colorWhite},
    #e6f4f4,
    ${colorBlueLight}
  );
  height: 280px;
  width: 340px;
  padding: 20px;
  box-shadow: 2px 2px 10px 2px ${rgba(colorGrayNormal, 0.7)};

  @media all and (min-width: 560px) {
    border-radius: 50px 0;
    height: 300px;
    width: 480px;
  }
`;

export const ModalText = styled.p`
  margin: 5px 0;
  font-family: ${fontSubtitle};
  font-size: 0.8rem;
  text-align: center;

  @media all and (min-width: 560px) {
    font-size: 1rem;
  }
`;

export const FlagsWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 70px;
  margin: 10px auto 0;

  @media all and (min-width: 560px) {
    margin: 20px auto 0;
  }
`;

export const Flag = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  outline: none;
  margin: 10px;
  cursor: pointer;

  ${props =>
    !props.active &&
    css`
      display: none;
    `}
`;

const activeFlag = css`
  opacity: 1;
  cursor: default;
`;

export const FlagLabel = styled.div`
  margin-bottom: 3px;
  font-family: ${fontTitleAlt};
  font-size: 0.8rem;
  text-align: center;
`;

export const FlagImage = styled.img`
  opacity: 0.5;
  margin-top: -10px;
  width: 100%;

  ${props =>
    props.lang === props.languageSelected &&
    css`
      ${activeFlag}
    `}
`;

export const CheckBox = styled.input`
  margin-right: 20px;
`;

export const CheckBoxArea = styled(ModalText)`
  margin: 30px 0 10px;
`;
