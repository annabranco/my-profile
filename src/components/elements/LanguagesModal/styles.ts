import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlueDark,
  colorBlack,
  colorGrayNormal,
  fontSubtitle,
  fontTitleAlt
} from '../../../styles/theme';
import { NotDisplayed } from '../../../styles/global';
import { ILanguageCode } from '../../../types/interfaces';

const activeFlag = css`
  opacity: 1;
  cursor: default;
`;

export const BackgroundOverlay = styled.div`
  z-index: 101;
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(
    to bottom,
    ${rgba(colorWhite, 0.6)},
    ${rgba(colorBlueDark, 0.4)},
    ${rgba(colorBlueDark, 0.6)},
    ${rgba(colorBlack, 0.9)}
  );
  height: ${`${window.innerHeight}px`};
  width: 100%;
`;
BackgroundOverlay.displayName = 'Background Overlay';

export const CheckBox = styled.input`
  margin-right: 20px;
`;
CheckBox.displayName = 'CheckBox';

export const Text = styled.p`
  margin: 5px 0;
  font-family: ${fontSubtitle};
  font-size: 0.8rem;
  text-align: center;

  @media all and (min-width: 560px) {
    font-size: 1rem;
  }
`;
Text.displayName = 'Text';

export const CheckBoxArea = styled(Text)`
  margin: 30px 0 10px;
`;
CheckBoxArea.displayName = 'CheckBox Area';

interface FlagProps extends React.ComponentPropsWithoutRef<'div'> {
  active?: boolean;
  languageSelected: ILanguageCode;
}

export const Flag = styled.div<FlagProps>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  outline: none;
  margin: 10px;
  cursor: pointer;

  ${({ active }) =>
    !active &&
    css`
      ${NotDisplayed};
    `}
`;
Flag.displayName = 'Flag';

export const FlagsWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 70px;
  margin: 10px auto 0;

  @media all and (min-width: 560px) {
    margin: 20px auto 0;
  }
`;
FlagsWrapper.displayName = 'Flags Wrapper';

interface ImageProps extends React.ComponentPropsWithoutRef<'img'> {
  lang: ILanguageCode;
  languageSelected: ILanguageCode;
}

export const Image = styled.img<ImageProps>`
  opacity: 0.5;
  margin-top: 0;
  width: 50%;

  ${({ lang, languageSelected }) =>
    lang === languageSelected &&
    css`
      ${activeFlag}
    `}
`;
Image.displayName = 'Image';

export const Label = styled.div`
  margin-bottom: 3px;
  font-family: ${fontTitleAlt};
  font-size: 0.8rem;
  text-align: center;
`;
Label.displayName = 'Label';

export const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border: 1px solid;
  border-radius: 40px 0;
  background-image: linear-gradient(
    to bottom left,
    ${colorWhite},
    #e6f4f4,
    ${colorBlueLight}
  );
  height: 280px;
  width: 340px;
  padding: 20px;
  box-shadow: 2px 2px 10px 2px ${rgba(colorGrayNormal, 0.7)};

  @media all and (min-width: 560px) {
    border-radius: 50px 0;
    height: 260px;
    width: 480px;
  }
`;
ModalWrapper.displayName = 'Modal Wrapper';
