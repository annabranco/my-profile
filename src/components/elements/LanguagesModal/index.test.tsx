import { mount, ReactWrapper } from 'enzyme';
import 'jest-styled-components';
import '../../../tests/mocks/matchMedia';
import LanguagesModal from '.';
import { setupMockProvider } from '../../../tests/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< LanguagesModal >', () => {
  let MockProvider;
  let wrapper: ReactWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;

    wrapper = mount(
      <MockProvider>
        <LanguagesModal
          hideForever
          onCloseLanguageModal={() => null}
          toggleBlockLangModal={() => null}
        />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should display all languages options received from server', () => {});

  xit('should display a flag and the name of each language', () => {});

  xit('should change the app language when the flag or name is clicked', () => {});

  xit('should close modal when OK button is clicked', () => {});

  xit('should set language preferencies on localstorage when OK button is clicked', () => {});
});
