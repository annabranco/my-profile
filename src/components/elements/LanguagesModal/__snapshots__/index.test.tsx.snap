// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< LanguagesModal > matches the snapshot 1`] = `
<MockProvider>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <LanguagesModal
      hideForever={true}
      onCloseLanguageModal={[Function]}
      toggleBlockLangModal={[Function]}
    >
      <Background Overlay>
        <div
          className="fWpqKC"
        >
          <Modal Wrapper
            data-e2e-id="languagesModal"
          >
            <div
              className="NKJtm"
              data-e2e-id="languagesModal"
            >
              <Flags Wrapper>
                <div
                  className="bwDZjO"
                >
                  <Flag
                    active={true}
                    aria-label="English"
                    data-e2e-id="langModal-flag_us"
                    key="en"
                    lang="en"
                    languageSelected="en"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      aria-label="English"
                      className="kBTZlT"
                      data-e2e-id="langModal-flag_us"
                      lang="en"
                      onClick={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <Label>
                        <div
                          className="dtIDQL"
                        >
                          English
                        </div>
                      </Label>
                      <Image
                        alt=""
                        lang="en"
                        languageSelected="en"
                        src="https://cdn.countryflags.com/thumbs/united-states-of-america/flag-3d-250.png"
                      >
                        <img
                          alt=""
                          className="dkDIFR"
                          lang="en"
                          src="https://cdn.countryflags.com/thumbs/united-states-of-america/flag-3d-250.png"
                        />
                      </Image>
                    </div>
                  </Flag>
                  <Flag
                    active={true}
                    aria-label="Español"
                    data-e2e-id="langModal-flag_es"
                    key="es"
                    lang="es"
                    languageSelected="en"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      aria-label="Español"
                      className="kBTZlT"
                      data-e2e-id="langModal-flag_es"
                      lang="es"
                      onClick={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <Label>
                        <div
                          className="dtIDQL"
                        >
                          Español
                        </div>
                      </Label>
                      <Image
                        alt=""
                        lang="es"
                        languageSelected="en"
                        src="https://cdn.countryflags.com/thumbs/spain/flag-3d-250.png"
                      >
                        <img
                          alt=""
                          className="fyzBXW"
                          lang="es"
                          src="https://cdn.countryflags.com/thumbs/spain/flag-3d-250.png"
                        />
                      </Image>
                    </div>
                  </Flag>
                  <Flag
                    active={true}
                    aria-label="Português"
                    data-e2e-id="langModal-flag_br"
                    key="pt"
                    lang="pt"
                    languageSelected="en"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      aria-label="Português"
                      className="kBTZlT"
                      data-e2e-id="langModal-flag_br"
                      lang="pt"
                      onClick={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <Label>
                        <div
                          className="dtIDQL"
                        >
                          Português
                        </div>
                      </Label>
                      <Image
                        alt=""
                        lang="pt"
                        languageSelected="en"
                        src="https://cdn.countryflags.com/thumbs/brazil/flag-3d-250.png"
                      >
                        <img
                          alt=""
                          className="fyzBXW"
                          lang="pt"
                          src="https://cdn.countryflags.com/thumbs/brazil/flag-3d-250.png"
                        />
                      </Image>
                    </div>
                  </Flag>
                  <Flag
                    active={false}
                    aria-label="Français"
                    data-e2e-id="langModal-flag_fr"
                    key="fr"
                    lang="fr"
                    languageSelected="en"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      aria-label="Français"
                      className="hjScPF"
                      data-e2e-id="langModal-flag_fr"
                      lang="fr"
                      onClick={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <Label>
                        <div
                          className="dtIDQL"
                        >
                          Français
                        </div>
                      </Label>
                      <Image
                        alt=""
                        lang="fr"
                        languageSelected="en"
                        src="https://cdn.countryflags.com/thumbs/france/flag-3d-250.png"
                      >
                        <img
                          alt=""
                          className="fyzBXW"
                          lang="fr"
                          src="https://cdn.countryflags.com/thumbs/france/flag-3d-250.png"
                        />
                      </Image>
                    </div>
                  </Flag>
                  <Flag
                    active={false}
                    aria-label="Пусский язык"
                    data-e2e-id="langModal-flag_ru"
                    key="ru"
                    lang="ru"
                    languageSelected="en"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      aria-label="Пусский язык"
                      className="hjScPF"
                      data-e2e-id="langModal-flag_ru"
                      lang="ru"
                      onClick={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <Label>
                        <div
                          className="dtIDQL"
                        >
                          Пусский язык
                        </div>
                      </Label>
                      <Image
                        alt=""
                        lang="ru"
                        languageSelected="en"
                        src="https://cdn.countryflags.com/thumbs/russia/flag-3d-250.png"
                      >
                        <img
                          alt=""
                          className="fyzBXW"
                          lang="ru"
                          src="https://cdn.countryflags.com/thumbs/russia/flag-3d-250.png"
                        />
                      </Image>
                    </div>
                  </Flag>
                  <Flag
                    active={false}
                    aria-label="Deutsch"
                    data-e2e-id="langModal-flag_de"
                    key="de"
                    lang="de"
                    languageSelected="en"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      aria-label="Deutsch"
                      className="hjScPF"
                      data-e2e-id="langModal-flag_de"
                      lang="de"
                      onClick={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <Label>
                        <div
                          className="dtIDQL"
                        >
                          Deutsch
                        </div>
                      </Label>
                      <Image
                        alt=""
                        lang="de"
                        languageSelected="en"
                        src="https://cdn.countryflags.com/thumbs/undefined/flag-3d-250.png"
                      >
                        <img
                          alt=""
                          className="fyzBXW"
                          lang="de"
                          src="https://cdn.countryflags.com/thumbs/undefined/flag-3d-250.png"
                        />
                      </Image>
                    </div>
                  </Flag>
                  <Flag
                    active={false}
                    aria-label="日本語"
                    data-e2e-id="langModal-flag_jp"
                    key="jp"
                    lang="jp"
                    languageSelected="en"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      aria-label="日本語"
                      className="hjScPF"
                      data-e2e-id="langModal-flag_jp"
                      lang="jp"
                      onClick={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <Label>
                        <div
                          className="dtIDQL"
                        >
                          日本語
                        </div>
                      </Label>
                      <Image
                        alt=""
                        lang="jp"
                        languageSelected="en"
                        src="https://cdn.countryflags.com/thumbs/undefined/flag-3d-250.png"
                      >
                        <img
                          alt=""
                          className="fyzBXW"
                          lang="jp"
                          src="https://cdn.countryflags.com/thumbs/undefined/flag-3d-250.png"
                        />
                      </Image>
                    </div>
                  </Flag>
                </div>
              </Flags Wrapper>
              <Text
                data-e2e-id="langModal-instructionsText"
              >
                <p
                  className="fzGfxz"
                  data-e2e-id="langModal-instructionsText"
                >
                  You may also change it in the upper bar
                </p>
              </Text>
              <CheckBox Area
                onClick={[Function]}
                role="checkbox"
              >
                <p
                  className="fzGfxz eFqbiK"
                  onClick={[Function]}
                  role="checkbox"
                >
                  <CheckBox
                    checked={true}
                    className="languages__checkbox"
                    data-e2e-id="langModal-checkbox"
                    id="checkLang"
                    name="hideForever"
                    readOnly={true}
                    type="checkbox"
                  >
                    <input
                      checked={true}
                      className="bNZqCL languages__checkbox"
                      data-e2e-id="langModal-checkbox"
                      id="checkLang"
                      name="hideForever"
                      readOnly={true}
                      type="checkbox"
                    />
                  </CheckBox>
                  <label
                    htmlFor="blockLang"
                  >
                    Do not show me this message again.
                  </label>
                </p>
              </CheckBox Area>
              <AppButton
                buttonStyle="confirm"
                datae2eid="langModal-ok"
                onClick={[Function]}
                text="ok"
              >
                <App Button
                  buttonStyle="confirm"
                  data-e2e-id="langModal-ok"
                  onClick={[Function]}
                  type="button"
                >
                  <button
                    className="dCozZD"
                    data-e2e-id="langModal-ok"
                    onClick={[Function]}
                    type="button"
                  >
                    ok
                  </button>
                </App Button>
              </AppButton>
            </div>
          </Modal Wrapper>
        </div>
      </Background Overlay>
    </LanguagesModal>
  </Provider>
</MockProvider>
`;
